### ISSUE TITLE
AFNI + cmdstanr

### ISSUE THREAD

**Issue by araikes** [NeuroDesk/neurocontainers#663]
Is it possible to have an AFNI container than bundles `cmdstanr`? `RBA` can take advantage of multi-core setups but only if cmdstanr is installed and available.

There's a docker file example [here ](https://github.com/JBris/stan-cmdstanr-docker/blob/main/Dockerfile)but it's built on rocker/tidyverse rather than fedora 35

**Comment by stebo85** [comment]
Dear @araikes,

Yes, that looks feasible. AFNI provides instructions for it:
https://afni.nimh.nih.gov/pub/dist/doc/program_help/RBA.html


Installation requirements: ~1~
In addition to R installation, the R package "brms" is required for RBA. Make
sure that you have the most recent version of R. To install "brms", run the following
command at the terminal:

rPkgsInstall -pkgs "brms" -site http://cran.us.r-project.org"/

Alternatively, you may install them in R:

install.packages("brms")

*** To take full advantage of parallelization, install both 'cmdstan' and 
'cmdstanr' and use the option -WCP in MBA. However, extra stpes are required: 
both 'cmdstan' and 'cmdstanr' have to be installed. To install 'cmdstanir',
execute the following command in R:

install.packages('cmdstanr', repos = c('[https://mc-stan.org/r-packages/',](https://mc-stan.org/r-packages/) getOption('repos')))

Then install 'cmdstan' using the following command in R:

cmdstanr::install_cmdstan(cores = 2)
# Follow the instruction here for the installation of 'cmdstan': 
#    https://mc-stan.org/cmdstanr/articles/cmdstanr.html
# If 'cmdstan' is installed in a directory other than home, use option -StanPath 
# to specify the path (e.g., -StanPath '~/my/stan/path').

In addition, if you want to show the ridge plots of the posterior distributions
through option -ridgePlot, make sure that the following R packages are installed:

data.table
ggplot2
ggridges
dplyr
tidyr
scales


Would you like to add this to the AFNI recipe here? https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/afni/build.sh 

Thank you
Steffen

**Comment by araikes** [comment]
@stebo85,

I will work on it.

**Comment by stebo85** [comment]
great :) Let me know if you need help. It should be a straight forward change hopefully!

**Comment by stebo85** [comment]
Dear @araikes - any updates on this?

**Comment by stebo85** [comment]
Dear @araikes - not sure how far you got with this, but I now included cmdstanr in the latest AFNI container (24.3.0) on Neurodesk. I tested the basic functions, but if you could test with your data and report back if it all works as expected that would be wonderful:

```
# Test cmdstanr
R
library(cmdstanr)
cmdstanr::set_cmdstan_path("/opt/cmdstan-2.35.0/")
file <- file.path(cmdstan_path(), "examples", "bernoulli", "bernoulli.stan")
mod <- cmdstan_model(file)
mod$print()
mod$exe_file()

data_list <- list(N = 10, y = c(0,1,0,0,0,0,0,0,0,1))

fit <- mod$sample(
  data = data_list,
  seed = 123,
  chains = 4,
  parallel_chains = 4,
  refresh = 500 # print update every 500 iters
)

```
